fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -135,8 +140,11 @@
 
     private static final PropertyConverter propertyConverter = new PropertyConverter();
 
-    private static Function<FedoraResource, Function<IdentifierConverter<Resource, FedoraResource>, Function<Boolean,
-            Stream<Triple>>>> getDefaultTriples = resource -> translator -> uncheck(minimal -> {
+    // A curried type accepting resource, translator, and "minimality", returning triples.
+    private static interface RdfGenerator extends Function<FedoraResource,
```

Comment:
Does it clarify the code or not? I feel like it does. In any event, if you aren't happy with it, I can definitely break it out into its own compilation unit and we can comment it up thoroughly. It might even provide part of a useful extension point that might get packed into the `kernel-api`.